packages/@aws-cdk/aws-s3-deployment/lib/lambda/index.py

```diff
@@ -114,8 +114,13 @@ def cfn_error(message=None):
 #---------------------------------------------------------------------------------------------------
 # populate all files from s3_source_zips to a destination bucket
 def s3_deploy(s3_source_zips, s3_dest, user_metadata, system_metadata, prune):
-    # create a temporary working directory
-    workdir=tempfile.mkdtemp()
+    # create a temporary working directory in /tmp or if enabled an attached efs volume
+    if 'EFS_ENABLED' in os.environ:
+        workdir = "/mnt/efs_tmp/" + str(uuid4())
+        os.mkdir(workdir)
```

Comment:
Should we delete this directory at the end? I imagine it has cost implications.

Reply:
workdir/content is killed, but yes this can also go.

Reply:
I will look for some time to update this PR and add a test next week

Reply:
shutil.rmtree(workdir) was already there.